File: app/src/main/java/com/android/app/fancytextreplacer/activities/MarkTextActivity.java
package com.android.app.fancytextreplacer.activities;

import android.app.Activity;
import android.os.Bundle;
import android.text.Html;
import android.view.MenuItem;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.android.app.fancytextreplacer.R;
import com.android.app.fancytextreplacer.utils.GlobalFunction;

import org.w3c.dom.Text;

public class MarkTextActivity extends AppCompatActivity {
    TextView tvOutput;
    String result;

    ImageView icBackOrMenu;
    TextView toolbarTitle;
    TextView tvnMarkText;
    LinearLayout outputLayout;
    EditText edtInput, edtTotalWord;
    GlobalFunction globalFunction;
    LinearLayout shareLayout, copyLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mark_text);

        icBackOrMenu = findViewById(R.id.icBackOrMenu);
        toolbarTitle = findViewById(R.id.toolbarTitle);
        toolbarTitle.setText("Mark Text");

        icBackOrMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });

        tvOutput = findViewById(R.id.tvOutput);
        edtInput = findViewById(R.id.edtInput);
        edtTotalWord = findViewById(R.id.edtTotalWord);
        tvnMarkText = findViewById(R.id.tvnMarkText);
        shareLayout = findViewById(R.id.shareLayout);
        copyLayout = findViewById(R.id.copyLayout);
        outputLayout = findViewById(R.id.outputLayout);
        globalFunction = new GlobalFunction(this);

        tvnMarkText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                hideKeyboard(MarkTextActivity.this);
                String input = edtInput.getText().toString();
                String num = edtTotalWord.getText().toString().trim();
                if (!input.isEmpty() && !num.isEmpty()) {
                    outputLayout.setVisibility(View.VISIBLE);
                    int totalWord = Integer.parseInt(num);
                    if (totalWord <= input.length()) {
                        String val = input.substring(0, totalWord);
                        String newS = input.replace(val, "");
                        String next = "<font color='#EE0000'>" + val + "</font>";
                        tvOutput.setText(Html.fromHtml(next + newS));
                        result = String.valueOf(Html.fromHtml(next + newS));
                    } else {
                        outputLayout.setVisibility(View.GONE);
                        Toast.makeText(MarkTextActivity.this, "Please donot enter limit greater than input length", Toast.LENGTH_SHORT).show();
                    }
                } else {
                    outputLayout.setVisibility(View.GONE);
                    Toast.makeText(MarkTextActivity.this, "Please enter input", Toast.LENGTH_SHORT).show();
                }
            }
        });

        shareLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!tvOutput.getText().toString().equals("Output")) {
                    globalFunction.shareMsg(tvOutput.getText().toString());
                } else {
                    Toast.makeText(getApplicationContext(), "No Output To Share", Toast.LENGTH_SHORT).show();
                }
            }
        });

        copyLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!tvOutput.getText().toString().equals("Output")) {
                    globalFunction.copyOutput(tvOutput.getText().toString());
                } else {
                    Toast.makeText(getApplicationContext(), "No Output To Share", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    public static void hideKeyboard(Activity activity) {
        InputMethodManager imm = (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
        View view = activity.getCurrentFocus();
        if (view == null) {
            view = new View(activity);
        }
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/stylish/fancy/text/generator/activities/MarkTextActivity.kt
package com.stylish.fancy.text.generator.activities

import android.app.Activity
import android.os.Bundle
import android.text.Html
import android.view.MenuItem
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.stylish.fancy.text.generator.R
import com.stylish.fancy.text.generator.databinding.ActivityMarkTextBinding
import com.stylish.fancy.text.generator.utils.GlobalFunction

class MarkTextActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMarkTextBinding

    var result: String? = null
    lateinit var icBackOrMenu: ImageView
    lateinit var toolbarTitle: TextView
    private lateinit var globalFunction: GlobalFunction

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_mark_text)

        icBackOrMenu = findViewById(R.id.icBackOrMenu)
        toolbarTitle = findViewById(R.id.toolbarTitle)
        toolbarTitle.text = getString(R.string.mark)
        globalFunction = GlobalFunction(this)
        icBackOrMenu.setOnClickListener { onBackPressed() }

        binding.tvnMarkText.setOnClickListener {
            hideKeyboard(this@MarkTextActivity)
            val input = binding.edtInput.text.toString()
            val num = binding.edtTotalWord.text.toString().trim { it <= ' ' }
            if (input.isNotEmpty() && num.isNotEmpty()) {
                val totalWord = num.toInt()
                if (totalWord <= input.length) {
                    binding.outputLayout.visibility = View.VISIBLE
                    binding.noOutput.visibility = View.GONE
                    val `val` = input.substring(0, totalWord)
                    val newS = input.replace(`val`, "")
                    val next = "<font color='#EE0000'>$`val`</font>"
                    binding.tvOutput.text = Html.fromHtml(next + newS)
                    result = Html.fromHtml(next + newS).toString()
                } else {
                    binding.noOutput.visibility = View.VISIBLE
                    binding.outputLayout.visibility = View.GONE
                    Toast.makeText(
                        this@MarkTextActivity,
                        "Please enter valid limit",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            } else {
                binding.outputLayout.visibility = View.GONE
                binding.noOutput.visibility = View.VISIBLE
                Toast.makeText(this@MarkTextActivity, "Please enter input", Toast.LENGTH_SHORT)
                    .show()
            }
        }

        binding.shareLayout.setOnClickListener {
            if (binding.tvOutput.text.toString() != "Output") {
                globalFunction.shareMsg(binding.tvOutput.text.toString())
            } else {
                Toast.makeText(applicationContext, "No Output To Share", Toast.LENGTH_SHORT).show()
            }
        }

        binding.copyLayout.setOnClickListener {
            if (binding.tvOutput.text.toString() != "Output") {
                globalFunction.copyOutput(binding.tvOutput.text.toString())
            } else {
                Toast.makeText(applicationContext, "No Output To Share", Toast.LENGTH_SHORT).show()
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        fun hideKeyboard(activity: Activity) {
            val imm = activity.getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
            var view = activity.currentFocus
            if (view == null) {
                view = View(activity)
            }
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }
}